File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        int maxc=0;
        for(int i=0;i<n;i++){
            set<int>set;
          for(int j=i;j<n;j++){

            if(set.find(s[j])!=set.end()){
                break;
            }
                          set.insert(s[j]);
           
            maxc=max(maxc,j-i+1);
          }
           
        }
        return maxc;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n=s.length();
        int maxc=0;
        for(int i=0;i<n;i++){
            Set<Character>set=new HashSet<>();
            for(int j=i;j<n;j++){
                if(set.contains(s.charAt(j))){
                    break;
                }
                set.add(s.charAt(j));
                maxc=Math.max(maxc,j-i+1);
            }
        }
        return maxc;
    }
}